### PR TITLE
[i18n-ar] Translated file : `docs/source/ar/model_memory_anatomy.md` into Arabic

### PR DESCRIPTION
**Pull request by AhmedAlmaghz**
## What does this PR do?
Translated the `docs/source/ar/model_memory_anatomy.md` file of the documentation to `Arabic العربية`. I use transformers for my side-project and wanted to contribute. As a beginner, this seemed appropriate.

### Some questions:

- I can't access the pr documentation endpoint for `Arabic العربية`. Is there a missing step?
- I left all the other files alone after copying the English docs. Is this acceptable for future updates or should I add contents to the `_toctree.yml` only when translation is complete? Should I delete all other files for a cleaner review?
- In `Arabic`, people use both `Transformer` and `المحولات` to describe the model. However, as this is more of a product of HuggingFace I opted not to translate it. May you please let me know which you would prefer?

Thank you in advance for your review.

Part of #32435

### Before submitting

- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests), Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/) ?Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

 ### Who can review?
 @stevhliu , @MKhalusova , @abodacs, may you please review this PR?

### CODE REVIEW

File: docs/source/ar/model_memory_anatomy.md [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+```
+الوقت: 57.82
+العينات / الثانية: 8.86
+ذاكرة GPU المشغولة: 14949 ميجابايت.
+```
+
+يمكننا أن نرى أن حجم دفعة صغير نسبيًا يملأ تقريبًا ذاكرة GPU بالكامل. ومع ذلك، غالبًا ما يؤدي حجم دفعة أكبر 
+في تقارب نموذج أسرع أو أداء أفضل في النهاية. لذلك نريد أن نضبط حجم الدفعة وفقًا لاحتياجات النموذج لدينا
+وليس قيود GPU. ما يثير الاهتمام هو أننا نستخدم ذاكرة أكثر بكثير من حجم النموذج. 
+لفهم سبب ذلك بشكل أفضل، دعنا نلقي نظرة على عمليات النموذج واحتياجاته من الذاكرة.
```
Comment (abodacs) [inline docs/source/ar/model_memory_anatomy.md:151]:
```suggestion
يمكننا أن نرى أن حجم دفعة صغير نسبيًا يملأ تقريبًا ذاكرة GPU بالكامل. ومع ذلك، غالبًا ما يؤدي حجم دفعة أكبر  في تقارب نموذج أسرع أو أداء أفضل في النهاية. لذلك نريد أن نضبط حجم الدفعة وفقًا لاحتياجات النموذج لدينا وليس  مع قيود وحدة GPU. ما يثير الاهتمام هو أننا نستخدم ذاكرة أكثر بكثير من حجم النموذج. 
لفهم سبب ذلك بشكل أفضل، دعنا نلقي نظرة على عمليات النموذج واحتياجاته من الذاكرة.
```

File: docs/source/ar/model_memory_anatomy.md [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+
+## تشريح عمليات النموذج
+
+تتضمن هندسة المحولات 3 مجموعات رئيسية من العمليات مجمعة أدناه حسب كثافة الحساب.
```
Comment (abodacs) [inline docs/source/ar/model_memory_anatomy.md:155]:
```suggestion
تتضمن بنية المحولات 3 مجموعات رئيسية من العمليات مُجمعة أدناه حسب كثافة العمليات الحسابية.
```

File: docs/source/ar/model_memory_anatomy.md [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+
+1. **التعاقدات المنسوجة**
```
Comment (abodacs) [inline docs/source/ar/model_memory_anatomy.md:157]:
```suggestion
1. **عمليات ضرب المصفوفات**
```

File: docs/source/ar/model_memory_anatomy.md [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+
+    تقوم الطبقات الخطية ومكونات الانتباه متعدد الرؤوس جميعها بعمليات **ضرب المصفوفة بالمصفوفة**. هذه العمليات هي الجزء الأكثر كثافة في الحساب من تدريب المحول.
```
Comment (abodacs) [inline docs/source/ar/model_memory_anatomy.md:159]:
```suggestion
    تقوم الطبقات الخطية ومكونات الانتباه متعدد الرؤوس جميعها بعمليات ضرب ** المصفوفة بالمصفوفة** على دفعات. هذه العمليات هي أكثر أجزاء تدريب المحولات كثافة من الناحية الحسابية.
```

File: docs/source/ar/model_memory_anatomy.md [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+
+2. **التطبيعات الإحصائية**
```
Comment (abodacs) [inline docs/source/ar/model_memory_anatomy.md:161]:
```suggestion
2. **عمليات التسوية الإحصائية**
```

File: docs/source/ar/model_memory_anatomy.md [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+
+    Softmax والتطبيع الطبقي أقل كثافة في الحساب من التعاقدات المنسوجة، وتنطوي على عملية أو أكثر من عمليات **الاختزال**، والتي يتم تطبيق نتيجتها بعد ذلك عبر خريطة.
```
Comment (abodacs) [inline docs/source/ar/model_memory_anatomy.md:163]:
```suggestion
تُعد  عمليات  Softmax والتسوية الطبقية أقل كثافة من ناحية  الحسابية من عمليات ضرب المصفوفات، وتنطوي على عملية أو أكثر من عمليات **الاختزال**، والتي يتم تطبيق نتيجتها بعد ذلك عبر خريطة.
```

File: docs/source/ar/model_memory_anatomy.md [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+
+3. **المشغلون عنصر الحكمة**
```
Comment (abodacs) [inline docs/source/ar/model_memory_anatomy.md:165]:
```suggestion
3. **العمليات على مستوى العناصر**
```

File: docs/source/ar/model_memory_anatomy.md [hunk 8]
```diff
@@ -0,0 +1,231 @@
...
+
+    هذه هي المشغلين المتبقية: **التحيزات، إسقاط، التنشيط، واتصالات بقايا**. هذه هي عمليات أقل كثافة في الحساب.
+
+يمكن أن تكون هذه المعرفة مفيدة لمعرفة عند تحليل اختناقات الأداء.
```
Comment (abodacs) [inline docs/source/ar/model_memory_anatomy.md:169]:
```suggestion
    هذه هي العمليات المتبقية: **الانحيازات، والتسرب، ووظائف التنشيط، والوصلات المتبقية**. هذه هي عمليات أقل كثافة من الناحية الحسابية.

يمكن أن تكون هذه المعرفة مفيدة لمعرفة عند تحليل اختناقات الأداء.
```

File: docs/source/ar/model_memory_anatomy.md [hunk 9]
```diff
@@ -0,0 +1,231 @@
...
+
+تم اشتقاق هذا الملخص من [Data Movement Is All You Need: A Case Study on Optimizing Transformers 2020](https://arxiv.org/abs/2007.00072)
```
Comment (abodacs) [inline docs/source/ar/model_memory_anatomy.md:171]:
```suggestion
هذا الملخص مُشتق من [نقل البيانات هو كل ما تحتاجه: دراسة حالة حول تحسين المحولات 2020](https://arxiv.org/abs/2007.00072)
```

File: docs/source/ar/model_memory_anatomy.md [hunk 10]
```diff
@@ -0,0 +1,231 @@
...
+
+لقد رأينا أن تدريب النموذج يستخدم ذاكرة أكثر بكثير من مجرد وضع النموذج على GPU. ويرجع ذلك إلى 
+هناك العديد من المكونات أثناء التدريب التي تستخدم ذاكرة GPU. المكونات الموجودة في ذاكرة GPU هي التالية:
+
+1. أوزان النموذج
+2. الدول المحسن
+3. التدرجات
+4. تنشيطات إلى الأمام المحفوظة لحساب التدرج
+5. المخازن المؤقتة المؤقتة
+6. ذاكرة محددة الوظائف
```
Comment (abodacs) [inline docs/source/ar/model_memory_anatomy.md:184]:
```suggestion
1. أوزان النموذج
2. الدول المُحسّن
3. المُتدرجات
4. تنشيطات المسار الأمامي المحفوظة لحساب  المُتدرجات
5. المخازن المؤقتة
6. ذاكرة محددة الوظائف
```

File: docs/source/ar/model_memory_anatomy.md [hunk 11]
```diff
@@ -0,0 +1,231 @@
...
+
+يتطلب نموذج نموذجي مدرب في الدقة المختلطة 18 بايت لكل معلمة نموذج بالإضافة إلى ذاكرة التنشيط. للاستدلال لا توجد حالات محسن وتدرجات، لذلك يمكننا طرح تلك. وهكذا ننتهي مع 6 بايت لكل 
+معلمة نموذج للدقة المختلطة الاستدلال، بالإضافة إلى ذاكرة التنشيط.
```
Comment (abodacs) [inline docs/source/ar/model_memory_anatomy.md:187]:
```suggestion
يتطلب نموذج نموذجي مدرب بدقة مختلطة 18 بايت للمُحسّن AdamW كل معلمة نموذج بالإضافة إلى ذاكرة التنشيط. للاستدلال لا توجد حالات مُحسّن و مُتدرجات، لذلك يمكننا طرح تلك. وهكذا ننتهي مع 6 بايت لكل 
معلمة نموذج للدقة المختلطة الاستدلال، بالإضافة إلى ذاكرة التنشيط.
```

File: docs/source/ar/model_memory_anatomy.md [hunk 12]
```diff
@@ -0,0 +1,231 @@
...
+
+دعنا نلقي نظرة على التفاصيل.
+
+**أوزان النموذج:**
+
+- 4 بايت * عدد المعلمات للتدريب fp32
+- 6 بايت * عدد المعلمات لتدريب الدقة المختلطة (يحافظ على نموذج في fp32 وواحد في fp16 في الذاكرة)
```
Comment (abodacs) [inline docs/source/ar/model_memory_anatomy.md:194]:
```suggestion
- 4 بايت * عدد المعلمات للتدريب على دقة fp32
- 6 بايت * عدد المعلمات لتدريب الدقة المختلطة (يحافظ على نموذج في fp32 وآخر بدقة fp16 في الذاكرة)
```

File: docs/source/ar/model_memory_anatomy.md [hunk 13]
```diff
@@ -0,0 +1,231 @@
...
+
+**حالات المحسن:**
```
Comment (abodacs) [inline docs/source/ar/model_memory_anatomy.md:196]:
```suggestion
**حالات المُحسّن:**
```

File: docs/source/ar/model_memory_anatomy.md [hunk 14]
```diff
@@ -0,0 +1,231 @@
...
+
+- 8 بايت * عدد المعلمات لمحسن AdamW العادي (يحافظ على حالتين)
+- 2 بايت * عدد المعلمات لمحسنات 8 بت AdamW مثل [bitsandbytes](https://github.com/TimDettmers/bitsandbytes)
+- 4 بايت * عدد المعلمات لمحسنات مثل SGD مع الزخم (يحافظ على حالة واحدة فقط)
```
Comment (abodacs) [inline docs/source/ar/model_memory_anatomy.md:200]:
```suggestion
- 8 بايت * عدد المعلمات للمُحسّن AdamW العادي (يحافظ على حالتين)
- 2 بايت * عدد المعلمات لمُحسّنات 8 بت AdamW مثل [bitsandbytes](https://github.com/TimDettmers/bitsandbytes)
- 4 بايت * عدد المعلمات لمُحسّنات مثل SGD مع الزخم momentum (يحافظ على حالة واحدة فقط)
```

File: docs/source/ar/model_memory_anatomy.md [hunk 15]
```diff
@@ -0,0 +1,231 @@
...
+
+**التدرجات**
```
Comment (abodacs) [inline docs/source/ar/model_memory_anatomy.md:202]:
```suggestion
**المُتدرجات**
```

File: docs/source/ar/model_memory_anatomy.md [hunk 16]
```diff
@@ -0,0 +1,231 @@
...
+
+- 4 بايت * عدد المعلمات للتدريب fp32 أو الدقة المختلطة (التدرجات تكون دائمًا في fp32)
```
Comment (abodacs) [inline docs/source/ar/model_memory_anatomy.md:204]:
```suggestion
- 4 بايت * عدد المعلمات للتدريب بدقة fp32 أو بدقة مختلطة (المُتدرجات تكون دائمًا بدقة fp32)
```

File: docs/source/ar/model_memory_anatomy.md [hunk 17]
```diff
@@ -0,0 +1,231 @@
...
+
+**تنشيطات إلى الأمام**
```
Comment (abodacs) [inline docs/source/ar/model_memory_anatomy.md:206]:
```suggestion
**تنشيطات المسار الأمامي**
```

File: docs/source/ar/model_memory_anatomy.md [hunk 18]
```diff
@@ -0,0 +1,231 @@
...
+
+- يعتمد الحجم على العديد من العوامل، وأهمها طول التسلسل وحجم المخفية وحجم الدفعة.
```
Comment (abodacs) [inline docs/source/ar/model_memory_anatomy.md:208]:
```suggestion
- يعتمد الحجم على العديد من العوامل، وأهمها طول التسلسل وحجم المخفية وحجم الدُفعة.
```

File: docs/source/ar/model_memory_anatomy.md [hunk 19]
```diff
@@ -0,0 +1,231 @@
...
+
+هناك الإدخال والإخراج الذي يتم تمريره وإعادته بواسطة وظائف إلى الأمام والخلف وتنشيطات إلى الأمام المحفوظة لحساب التدرج.
```
Comment (abodacs) [inline docs/source/ar/model_memory_anatomy.md:210]:
```suggestion
هناك  المدخلات والمخرجات لذي يتم تمريرها وإرجاعها بواسطة وظائف المسار  الأمامي والمسار الخلفي وتنشيطات المسار الأمامي المحفوظة لحساب المُتدرجات.
```

File: docs/source/ar/model_memory_anatomy.md [hunk 20]
```diff
@@ -0,0 +1,231 @@
...
+
+**الذاكرة المؤقتة**
+
+بالإضافة إلى ذلك، هناك جميع أنواع المتغيرات المؤقتة التي يتم تحريرها بمجرد الانتهاء من الحساب، ولكن في 
+لحظة يمكن أن تتطلب هذه المتغيرات المؤقتة ذاكرة إضافية ويمكن أن تدفع إلى OOM. لذلك، عند الترميز، من المهم التفكير 
+بشكل استراتيجي حول هذه المتغيرات المؤقتة وأحيانًا تحريرها بشكل صريح بمجرد عدم الحاجة إليها.
```
Comment (abodacs) [inline docs/source/ar/model_memory_anatomy.md:216]:
```suggestion
بالإضافة إلى ذلك، هناك جميع أنواع المتغيرات المؤقتة التي يتم تحريرها بمجرد الانتهاء من الحساب، ولكن في 
لحظة يمكن أن تتطلب هذه المتغيرات المؤقتة ذاكرة إضافية ويقد تؤدي إلى نفاد  الذاكرة المُخصصة (OOM). لذلك، عند البرمجة، من المهم التفكير بشكل استراتيجي حول هذه المتغيرات المؤقتة وأحيانًا تحريرها بشكل صريح بمجرد عدم الحاجة إليها.
```

File: docs/source/ar/model_memory_anatomy.md [hunk 21]
```diff
@@ -0,0 +1,231 @@
...
+
+**ذاكرة محددة الوظائف**
+
+ثم، قد يكون لبرنامجك متطلبات ذاكرة خاصة. على سبيل المثال، عند إنشاء نص باستخدام البحث الشعاعي، يحتاج البرنامج
```
Comment (abodacs) [inline docs/source/ar/model_memory_anatomy.md:220]:
```suggestion
ثم، قد يكون لبرنامجك احتياجات خاصة بالذاكرة. على سبيل المثال، عند إنشاء نص باستخدام البحث الشعاعي، يحتاج البرنامج 
```

File: docs/source/ar/model_memory_anatomy.md [hunk 22]
```diff
@@ -0,0 +1,231 @@
...
+**الذاكرة المؤقتة**
+
+بالإضافة إلى ذلك، هناك جميع أنواع المتغيرات المؤقتة التي يتم تحريرها بمجرد الانتهاء من الحساب، ولكن في 
+لحظة يمكن أن تتطلب هذه المتغيرات المؤقتة ذاكرة إضافية ويمكن أن تدفع إلى OOM. لذلك، عند الترميز، من المهم التفكير 
+بشكل استراتيجي حول هذه المتغيرات المؤقتة وأحيانًا تحريرها بشكل صريح بمجرد عدم الحاجة إليها.
+
+**ذاكرة محددة الوظائف**
+
+ثم، قد يكون لبرنامجك متطلبات ذاكرة خاصة. على سبيل المثال، عند إنشاء نص باستخدام البحث الشعاعي، يحتاج البرنامج 
+للحفاظ على عدة نسخ من الإدخالات والمخرجات.
```
Comment (abodacs) [inline docs/source/ar/model_memory_anatomy.md:221]:
```suggestion
إلى الاحتفاظ بنسخ متعددة من المدخلات والمخرجات.
```

File: docs/source/ar/model_memory_anatomy.md [hunk 23]
```diff
@@ -0,0 +1,231 @@
...
+
+**سرعة تنفيذ `forward` مقابل `backward`**
+
+بالنسبة للضربات والطبقات الخطية، هناك 2x flops في الخلف مقارنة بالخط الأمامي، والتي تترجم عمومًا إلى ~2x أبطأ (أحيانًا أكثر، لأن الأحجام في الخلف تميل إلى أن تكون أكثر إزعاجًا). عادةً ما تكون التنشيطات محدودة النطاق، ومن المعتاد أن يتعين على التنشيط قراءة المزيد من البيانات في الخلف أكثر من الأمام
```
Comment (abodacs) [inline docs/source/ar/model_memory_anatomy.md:225]:
```suggestion
بالنسبة للالتفافات والطبقات الخطية، هناك ضِعف عدد العمليات 2x flops في المسار الخلفى مقارنة بالمسار الأمامي، والتي يُترجم عمومًا إلى ~2x أبطأ (أحيانًا أكثر، لأن الأحجام في المسار الخلفى تميل إلى أن تكون أكثر صعوبة). عادةً ما تكون عمليات التنشيط محدودة بعرض  النطاق  الترددي، ومن المعتاد أن يتعين على التنشيط قراءة المزيد من البيانات في المسار الخلفى أكثر من المسار الأمامى.
```

File: docs/source/ar/model_memory_anatomy.md [hunk 24]
```diff
@@ -0,0 +1,231 @@
...
+**ذاكرة محددة الوظائف**
+
+ثم، قد يكون لبرنامجك متطلبات ذاكرة خاصة. على سبيل المثال، عند إنشاء نص باستخدام البحث الشعاعي، يحتاج البرنامج 
+للحفاظ على عدة نسخ من الإدخالات والمخرجات.
+
+**سرعة تنفيذ `forward` مقابل `backward`**
+
+بالنسبة للضربات والطبقات الخطية، هناك 2x flops في الخلف مقارنة بالخط الأمامي، والتي تترجم عمومًا إلى ~2x أبطأ (أحيانًا أكثر، لأن الأحجام في الخلف تميل إلى أن تكون أكثر إزعاجًا). عادةً ما تكون التنشيطات محدودة النطاق، ومن المعتاد أن يتعين على التنشيط قراءة المزيد من البيانات في الخلف أكثر من الأمام 
+(على سبيل المثال، قراءة التنشيط إلى الأمام مرة واحدة، والكتابة مرة واحدة، وقراءة التنشيط الخلفي مرتين، gradOutput وإخراج الأمام، 
+والكتابة مرة واحدة، gradInput).
```
Comment (abodacs) [inline docs/source/ar/model_memory_anatomy.md:227]:
```suggestion
(على سبيل المثال، قراءة التنشيط المسار الأمامى مرة واحدة، وتكتب مرة واحدة، وبينما  تقرأ  عملية التنشيط الخلفي مرتين، gradOutput وإخراج الأمام، وتكتب مرة واحدة، gradInput).
```

**Review by abodacs — CHANGES_REQUESTED** [review]
السﻻم عليكم

@AhmedAlmaghz 
تمت المراجعة 

جزاء الله خيرا

**Comment by AhmedAlmaghz** [comment]
Thank you @abodacs

**Comment by AhmedAlmaghz** [comment]
اشكرك اخي عبدالله @abodacs  على جهودك الرائعة.

عمل رائع وعظيم 

التالي (الأخير في القائمة الحالية) : 🥇 

# CONCEPTUAL GUIDES

- [x]  Philosophy #33064
- [x]  Glossary #33038
- [x]  What 🤗 Transformers can do #33073
- [x]  How 🤗 Transformers solve tasks #33074
- [x]  The Transformer model family #33047
- [x]  Summary of the tokenizers #33078
- [ ]  Attention #33021
- [ ]  Padding and truncation #33050
- [ ]  BERTology #33024
- [ ]  Perplexity of fixed-length models #33063
- [ ]  Pipelines for webserver inference #33066
- [ ]  Model training anatomy #33045
- [ ] Getting the most out of LLMs #33043

**Review by abodacs — APPROVED** [review]
@AhmedAlmaghz
برجاء معالجة التعارض

**Pull request closed without merging**